validation/views.py

```diff
@@ -707,6 +734,59 @@ def save_wrong_word(request, recording_id):
     return response
 
 
+@login_required()
+def record_audio(request):
+    if request.method == "POST":
+        form = RecordNewPhrase(request.POST, request.FILES)
+        translation = request.POST.get("translation")
+        transcription = request.POST.get("transcription")
+        audio_data = request.FILES["audio_data"]
+
+        phrase = Phrase.objects.filter(transcription=transcription).first()
+        if not phrase:
+            phrase = Phrase(
+                translation=translation,
+                field_transcription=transcription,
+                transcription=transcription,
+                kind=Phrase.SENTENCE if " " in transcription else Phrase.WORD,
+                status=Phrase.USER,
+                date=datetime.datetime.now(),
+                modifier=request.user,
+            )
+            phrase.save()
+
+        speaker = Speaker.objects.filter(user=request.user).first()
+        if not speaker:
+            speaker = Speaker(
+                full_name=request.user.first_name + " " + request.user.last_name,
+                code=request.user.first_name[:3].upper(),
+            )
+            speaker.save()
+
+        _id = create_new_rec_id(phrase, speaker)
+        audio_data.name = _id + ".wav"
+        rec = Recording(
+            id=_id,
+            compressed_audio=audio_data,
+            speaker=speaker,
+            phrase=phrase,
+            timestamp=0,
+            comment=f"Uploaded by user: {request.user}",
+            is_user_submitted=True,
+        )
+        rec.save()
+        source = settings.RECVAL_AUDIO_PREFIX + _id + ".wav"
+        dest = settings.RECVAL_AUDIO_PREFIX + _id + ".m4a"
+        subprocess.check_call(["ffmpeg", "-i", source, dest], cwd=settings.MEDIA_ROOT)
+        rec.compressed_audio = dest
+        rec.save()
+        os.remove(settings.MEDIA_ROOT + "/" + source)
```

Comment:
No harm in keeping the higher-quality original, just takes a bit of extra disk space.
```suggestion
```